### DISCUSSION TITLE
Please advise what am i doing wrong here ...

### DISCUSSION THREAD

**Discussion by udiazulayintel** [rabbitmq/rabbitmq-dotnet-client#1395] [Q&A]
Hi,

I am having an issue like this :
A publisher publishes X messages to a Q, say X=2,345,678 (issue occurs on large amount ...)
When A consumer is consuming the massages after the publisher completed - all X messages processed fine!

When the consumer is consuming the massages while the publisher is publishing, the consumer is processing less messages ...

One important thing to mention : i need to process batch (so i can push a batch to SQL Server) - so i am collecting the messages in a Data Table and a timer is pushing them to the DB and Ack the message, clearing the table etc.

If i am acknowledging the messages right after i add them to the data table - all works well, here is a snapshot of my code :

``` 
  ulong lastDeliveryTag = 0;

  consumer.Received += (ch, ea) =>
  {  
        lock (channelLock)
        {
            var body = ea.Body.ToArray();
            
            // Convert byte array to string using UTF-8 encoding
            string resultString = Encoding.UTF8.GetString(body);

            AddLogItemToTable(dataTable, resultString, testerName, cellName, channelLock);                            
            
            lastDeliveryTag = ea.DeliveryTag;  
                                 
            ....
```

Timer Code :

```
  _ = Task.Run(async () =>
  {
    while (true)
    {
        await Task.Delay(_siteConfiguration.TaskTimerInMs);
        // This code process the data table to the DB and clean...
        
        BulkCopyToDatabase();   
        channel.BasicAck(lastDeliveryTag, multiple: true);                         
    }     

  });
```

Please advise what am i doing wrong here ... ?

Thanks!!

**Comment by lukebakken** [comment]
Thanks for using RabbitMQ and the .NET client.

It is very difficult to make suggestions based on short, incomplete code snippets. I can't really tell what you're trying to do in your `Received` handler.

If you would like us to assist you, you should share two console applications - one that publishes messages, and one that consumes them, that demonstrates the issue you are having. Please share them via a git repository (via GitHub or GitLab). I should be able to clone it, and run `dotnet run` to see the behavior.
